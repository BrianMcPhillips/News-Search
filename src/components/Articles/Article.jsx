import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ 
  title, description, author, publishedAt, 
  url, imageUrl, content 
}) => (
  <>
    <h2>{title}</h2>
    <h3>{description}</h3>
    <h4>{author}</h4>
    <p>{publishedAt}</p>
    <p>{url}</p>
    <img src={imageUrl} alt={title} />
    <p>{content}</p>
  </>
);

Article.propTypes = {
  
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  publishedAt: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired
  
  
};

export default Article;
