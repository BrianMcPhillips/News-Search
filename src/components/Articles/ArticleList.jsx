import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';



const ArticleList = ({ articles }) => {
  const articleElements = articles.map(article => (
    <Article 
      title={article.title}
      description={article.description}
      author={article.author}
      publishedAt={article.publishedAt}
      key={article.title}
    />
  ));

  return (
    <div>
      <h1>Articles:</h1>
      {articleElements}
    </div>
  );
};

ArticleList.propTypes = {
  articleList: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    publishedAt: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    imageUrl: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired
  })
  ),
};

export default ArticleList;
