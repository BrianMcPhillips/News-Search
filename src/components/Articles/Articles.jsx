import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';



const Articles = ({ articles }) => {
  const articleElements = articles.map(article => (
    <Article 
      title={article.title}
      description={article.description}
      author={article.author}
      publishedAt={article.publishedAt}
      url={article.url}
      imageUrl={article.imageUrl}
      content={article.content}
      key={article.title}
    />
  ));

  return (
    <ul data-testid="articles">
      {articleElements}
    </ul>
  );
};

Articles.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    publishedAt: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    imageUrl: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired
  })
  ),
};

export default Articles;
