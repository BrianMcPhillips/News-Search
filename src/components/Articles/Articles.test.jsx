import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Articles from './Articles';

describe('Articles component', () => {
  afterEach(() => cleanup());
  it('renders all Articles', () => {
    const { asFragment } = render(<Articles
      articles={[
        {
          title: 'Test Article',
          description: 'This is a test description',
          author: 'Test Testerson',
          publishedAt: '12/12/2020',
          url: 'testurl.com/test/test',
          imageUrl: 'http://placeholder.com/article',
          content: `This is the best test article ever written, voted best
            articleby the New York Times and associated who may or may not 
            exist`
        },
        {
          title: 'Test Article 2',
          description: 'This is the second test description',
          author: 'Test Testerson Jr.',
          publishedAt: '12/14/2020',
          url: 'testurl.com/test/test2',
          imageUrl: 'http://placeholder.com/article2',
          content: `This is the second best test article ever written, voted 
          second best articleby the New York Times and associated who may or 
          may not exist`
        }
      ]}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
