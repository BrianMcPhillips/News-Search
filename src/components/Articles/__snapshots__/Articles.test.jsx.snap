// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles component renders all Articles 1`] = `
<DocumentFragment>
  <ul
    data-testid="articles"
  >
    <h2>
      Test Article
    </h2>
    <h3>
      This is a test description
    </h3>
    <h4>
      Test Testerson
    </h4>
    <p>
      12/12/2020
    </p>
    <p>
      testurl.com/test/test
    </p>
    <img
      alt="Test Article"
      src="http://placeholder.com/article"
    />
    <p>
      This is the best test article ever written, voted best
            articleby the New York Times and associated who may or may not 
            exist
    </p>
    <h2>
      Test Article 2
    </h2>
    <h3>
      This is the second test description
    </h3>
    <h4>
      Test Testerson Jr.
    </h4>
    <p>
      12/14/2020
    </p>
    <p>
      testurl.com/test/test2
    </p>
    <img
      alt="Test Article 2"
      src="http://placeholder.com/article2"
    />
    <p>
      This is the second best test article ever written, voted 
          second best articleby the New York Times and associated who may or 
          may not exist
    </p>
  </ul>
</DocumentFragment>
`;
